integr8ly/integreatly-operator
# Add rate limit to backend-listener

# Description

We needed to apply the rate-limiting settings to the backend-listener to enable the usage of self-managed APICast, this PR also refactors the 3scale reconciler by moving many parts of rate limit resources creation to other package and go files.

## Type of change



# Verification steps

1. Install RHOAM from the branch 
2. When the install is complete make sure the backend-listener pod has also an envoy sidecar container
3. Verify if an EnvoyConfig CR was created for the backend-listener
4. Ensure the backend-listener-proxy service was created in the 3scale namespace and the backend-listener route points to this service
5. Stop the operator and change the ratelimit configmap in the marin3r namespace to allow 2 requests
6. Curl the backend-listener external route one time, then curl the APICast route two times, when the APICast route responds to the second request, it should return `429 - too many requests`
